orbmain/pom.xml

```
@@ -88,6 +88,11 @@
             <scope>test</scope>
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <groupId>org.glassfish.gmbal</groupId>
+            <artifactId>gmbal</artifactId>
+            <scope>test</scope>
```

I don't understand - why tests here do not fail without that dependency?

---

It depends on gmbal, which in older released artifacts swallowed exceptions.

---

I cannot see _failures in tests_ here before.

Before:
```
[WARNING] Tests run: 328, Failures: 0, Errors: 0, Skipped: 2
```
```
[INFO] --- dependency:3.7.0:analyze-only (default-cli) @ glassfish-corba-orb ---
[WARNING] Used undeclared dependencies found:
[WARNING]    org.glassfish.external:management-api:jar:3.3.0:compile
[WARNING] Unused declared dependencies found:
[WARNING]    org.glassfish.corba:glassfish-corba-csiv2-idl:jar:5.0.0-SNAPSHOT:compile
[WARNING]    javax.transaction:javax.transaction-api:jar:1.3:test
[WARNING]    org.glassfish.corba:exception-annotation-processor:jar:5.0.0-SNAPSHOT:provided
```

with this added:
```
[WARNING] Tests run: 328, Failures: 0, Errors: 0, Skipped: 2
```
```
[INFO] --- dependency:3.7.0:analyze-only (default-cli) @ glassfish-corba-orb ---
[WARNING] Used undeclared dependencies found:
[WARNING]    org.glassfish.external:management-api:jar:3.3.0:compile
[WARNING] Unused declared dependencies found:
[WARNING]    org.glassfish.corba:glassfish-corba-csiv2-idl:jar:5.0.0-SNAPSHOT:compile
[WARNING]    javax.transaction:javax.transaction-api:jar:1.3:test
[WARNING]    org.glassfish.gmbal:gmbal:jar:4.1.0:test
[WARNING]    org.glassfish.corba:exception-annotation-processor:jar:5.0.0-SNAPSHOT:provided
```

This addition is unclear for me.

Could _failures in tests_ be clarified? How can I see failures before this change and them being gone after?

---

> I don't understand - why tests **here** do not fail without that dependency?

Can you first approve these two? I will restage them, then master here should fail:

Why they did not fail here but later GF tests failed - I added some comments here to make it clearer: 
* https://github.com/eclipse-ee4j/orb-gmbal/pull/56/files
* Here, when the dependency was not available, the GenericConstructor just logged the error and returned null.
* static methods in `ManagedObjectManagerFactory` then returned `ManagedObjectManagerNOPImpl.self` which passed local tests this way, but is worthless in real application.

---

Master still uses dependencies staged before this. I will probably drop them now, as they are not ready to be released. We should target the exception swallowing more. It is easier to fix something when it says what happened, than having to enable extremely verbose logging and then dig in the output.

I have also doubts about the NOOP impl, but now we cannot start refactoring whole ORB, there are higher priorities.

---

> Can you first approve these two? I will restage them, then master here should fail:

Thanks for explanation.

Now I see - I've chosen the wrong order of changes to review.

---

With 

- https://github.com/eclipse-ee4j/orb-gmbal-pfl/pull/94
- https://github.com/eclipse-ee4j/orb-gmbal/pull/56

rebuilt and locally installed as their target release (not -SNAPSHOTs) - and now I see the failures in `orbmain`.